src/arch/riscv/kernel/context.c

```diff
@@ -1,11 +1,15 @@
-#include <kernel/panic.h>
+#include <stdint.h>
 #include <hal/context.h>
+#include <hal/ipl.h>
+#include <asm/regs.h>
 
 void context_init(struct context *ctx, unsigned int flags,
 		void (*routine_fn)(void), void *sp) {
```

Comment:
I think that unused fields of `ctx` should be set to zero in this function to make sure there are no "garbage" in the register.
You can use `memset(ctx, 0, sizeof(*ctx))` at the beginning of the function.